standard-libraries/iterator.3.mr

```diff
@@ -0,0 +1,15 @@
+## Generic iterator dynamic interface
+
+
+class Iterator{Generic}
+  ## Return whether current item exist
+  func dynamic has-item():(copy Bool exist)
```

Comment:
You misunderstood this method, it doesn't search - it is the stop condition of the iterator - when it returns `false` it means the iteration ended.
This proves it's has a bad name (and documentation) - I should rename it to `is-end` or something like that, what do you suggest?